testsuite/features/support/code_coverage.rb

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2016-2023 SUSE LLC.
+# Licensed under the terms of the MIT license.
+require 'redis'
+require 'nokogiri'
+require 'open-uri'
+
+# CodeCoverage handler to produce, parse and report Code Coverage from the Jave Server to our GitHub PRs
+class CodeCoverage
+  include(Nokogiri::XML)
+  ##
+  # Initialize a connection with a Redis database
+  def initialize(redis_host, redis_port, redis_username, redis_password)
+    @database = Redis.new(host: redis_host, port: redis_port, username: redis_username, password: redis_password)
+  end
+
+  ##
+  # Close the connection with the Redis database
+  def close
+    @database.close
+  end
+
+  ##
+  # Parse a Jacoco XML report, extracting information that will be included in a Set on a Redis database
+  #
+  # Args:
+  #   feature_name: The feature name of the code coverage report we want to parse and push to Redis
+  def push_feature_coverage(feature_name)
+    filename = "/tmp/jacoco-#{feature_name}.xml"
+    begin
+      xml = File.read(filename, mode: 'r')
+      options = Nokogiri::XML::ParseOptions.new(ParseOptions::HUGE | ParseOptions::RECOVER | ParseOptions::NONET)
+      tree = Nokogiri::XML::Document.parse(xml, nil, nil, options)
+      tree.xpath('.//package').each do |package|
+        package_name = package.attr('name')
+        package.xpath('.//sourcefile').each do |sourcefile|
+          sourcefile_name = sourcefile.attr('name')
+          counter_class = sourcefile.xpath(".//counter[@type='CLASS']")
+          next if counter_class.nil? || counter_class.attr('covered').to_s == ''
+
+          next unless Integer(counter_class.attr('covered').to_s).positive?
+
+          begin
+            @database.sadd("#{package_name}/#{sourcefile_name}", feature_name)
+          rescue StandardError => e
+            warn("#{e.backtrace} > #{package_name}/#{sourcefile_name} : #{feature_name}")
+          end
+        end
+      end
+    rescue StandardError => e
+      warn(e.backtrace)
+    ensure
+      File.delete(filename)
+    end
+  end
+
+  ##
+  # Parse a Jacoco XML report, extracting information that will be included in a Set on a Redis database
```

Comment:
This is a duplicate. You already have this description above in line 23. Furthermore it does not describe `jacoco_dump()`

Reply:
Right. I rushed too much. Good catch.